content/docs/atom-guide/look-dev/vegetation-bending-tutorial.md

```diff
@@ -0,0 +1,725 @@
+   {{< note >}}
+   This function could potentially be run once per-object on the CPU with the results updated each frame in the ObjectSrg instead of being run once per-vertex on the GPU. It's a tradeoff between doing the computation once per frame per vegetation object + an extra SRG compile per frame per vegetation object vs. re-doing the computation on the GPU per frame per vegetation object per vertex. It's content specific, with the redundant GPU cost increasing as vertex density increases, and also depends on if the GPU is the bottleneck, and whether or not the vertex shader is the bottleneck, and also if the vertex shader is bandwidth bound or ALU bound. 
+   {{< /note >}}
+3. Now, let's call the function in our `ProceessBending` function. Inside the conditional, add a call to our wind function:
+   
+   ```hlsl
+   if (o_color_isBound) 
+   {
+      // Overall wind
+      float4 currentBending = SetUpWindBending(currentTime, worldPosition);
+   }
+
+   return adjustedWorldPosition;
+   ```
+
+Great, now we have our wind bending function set up! Note that this doesn't enact any changes on our tree just yet, and the tree should be rendered as normal.
+
+### Add detail bending
+Using the wind bending constants that we just calculated, we can now determine the bending of the leaves.
+
+1. Open `{your-project-path}\Materials\Types\VegetationBending_Common.azsli`.
+2. At the bottom, add the following code:
+   
+   ```
+   float3 DetailBending(float3 objectSpacePosition, float3 normal, float4 color, float currentTime, float4 worldPosition, float bendLength)
+   {
+      // The information from the vertex colors about how to bend this vertex.
+      float edgeInfo = color.x;
+      float branchPhase = color.y;
+      float branchBendAmount = color.z;
+
+      // Phases (object, vertex, branch)
+      float objPhase = dot(worldPosition.xyz, 2.0); 
+      branchPhase += objPhase;
+      float vtxPhase = dot(objectSpacePosition, branchPhase); 
+
+      // Detail bending for leaves
+      // x: is used for leaves, y is used for branch
+      float2 wavesIn = currentTime;
+      wavesIn += float2(vtxPhase, branchPhase);
+      float4 waves = (frac(wavesIn.xxyy * float4(1.975, 0.793, 0.375,  0.193)) * 2.0 - 1.0) * MaterialSrg::m_detailFrequency * bendLength;
+      waves = abs(frac(waves + 0.5) * 2.0 - 1.0);
+
+      // x: is used for leaves, y is used for branches
+      float2 wavesSum = waves.xz + waves.yw;
+
+      // Leaf and branch bending (xy is used for leaves, z for branches)
+      float3 movement = wavesSum.xxy * float3(edgeInfo * MaterialSrg::m_detailLeafAmplitude * normal.xy, branchBendAmount * MaterialSrg::m_detailBranchAmplitude);
+      return objectSpacePosition + movement;
+   }
+   ```
+3. Call the detail bending function in our `ProceessBending` function. Inside the conditional, add the call:
```

Comment:
The ProcessBending function as shown above is already calling DetailBending, so this is a little confusing.